core/src/main/scala/Logic.scala

```diff
@@ -13,58 +13,57 @@ class CarApi(carService: CarService):
       input.split(" ", 3).toList match
         case List(f1, f2, f3) =>
           val car = Car(f1, f2, f3)
-          carService.register(car).map(_ => "OK: Car registered").catchAll {
+          carService.register(car).as("OK: Car registered").catchAll {
             case _: LicensePlateExistsError =>
               ZIO
                 .logError(
                   s"Cannot register: $car, because a car with the same license plate already exists"
                 )
-                .map(_ => "Bad request: duplicate")
+                .as("Bad request: duplicate")
             case t =>
               ZIO
                 .logErrorCause(s"Cannot register: $car, unknown error", Cause.fail(t))
-                .map(_ => "Internal server error")
+                .as("Internal server error")
           }
-        case _ => ZIO.logError(s"Bad request: $input").map(_ => "Bad Request")
+        case _ => ZIO.logError(s"Bad request: $input").as("Bad Request")
     }
 
 object CarApi:
-  val live: ZLayer[CarService, Any, CarApi] = ZLayer.fromFunction(CarApi(_))
+  val live: ZLayer[CarService, Nothing, CarApi] = ZLayer.fromFunction(CarApi(_))
+  def register(input: String): ZIO[CarApi, Nothing, String] =
+    ZIO.serviceWithZIO[CarApi](_.register(input))
 
 //
 
-class CarService(carRepository: CarRepository, db: DB):
+class CarService(db: DB):
   def register(car: Car): ZIO[Any, Throwable | LicensePlateExistsError, Unit] =
     db.transact {
-      carRepository.exists(car.licensePlate).flatMap {
+      CarRepository.exists(car.licensePlate).flatMap {
         case true  => ZIO.fail(LicensePlateExistsError(car.licensePlate))
-        case false => carRepository.insert(car)
+        case false => CarRepository.insert(car)
       }
     }
 
 object CarService:
-  val live: ZLayer[CarRepository & DB, Nothing, CarService] =
-    ZLayer.fromFunction(CarService(_, _))
+  val live: ZLayer[DB, Nothing, CarService] =
+    ZLayer.fromFunction(CarService(_))
 
 //
 
-class CarRepository():
+object CarRepository:
```

Comment:
I would prefer to keep this as a class for two reasons: (1) regularity, (2) in a testing environment, you might want to substitute that particular component